### PR TITLE
New style submenu

### PR DESCRIPTION
**Pull request by vxsx**
This is based on https://github.com/divio/django-cms/pull/4453

| before | after |
|----------|-------|
|screen shot 2015-09-16 at 17 34 04 | screen shot 2015-09-16 at 17 33 13 screen shot 2015-09-16 at 17 33 30  screen shot 2015-09-16 at 18 11 57 |

buttons positioning trust last screenshot

### CODE REVIEW

File: cms/static/cms/js/modules/cms.plugins.js [hunk 1]
```diff
@@ -509,7 +514,82 @@
                     e.stopPropagation();
                 });
 
-                dropdown.find('a').on('click.cms', function (e) {
+                that._setupActions(nav);
+                // prevent propagnation
+                nav.on(this.click + ' pointerup.cms pointerdown.cms click.cms dblclick.cms', function (e) {
```
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.plugins.js]: please keep in mind on cleanup:

- Move all elements to ``this.ui`` (search for .find() and replace)
- add YUIdocs to general outline
- set the name for anonymous functions: ``_events: function () {`` becomes ``_events: function _events() {``
- set event handler variables like ``this.pointerup = 'pointerup.cms.sideframe pointercancel.cms';``
- replace ``bind`` with ``on`` and ``unbind`` with ``off``
- ensure for correct variable name positioning
- remove as many timeouts as possible

File: cms/static/cms/js/modules/cms.plugins.js [hunk 2]
```diff
@@ -509,7 +514,82 @@
...
+                });
+
+                that._setupQuickSearch(nav);
+            },
+
+            /**
+             * sets up event handlers for quicksearching
+             * FIXME will be moved into a separate "add plugin" modal
+             *
+             * @param nav jQuery
```
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.plugins.js]: @private

File: cms/static/cms/sass/components/_subnav.scss [hunk 3]
```diff
@@ -98,6 +134,103 @@
...
+        box-shadow: $cms-dropdown-shadow;
+        right: $submenu-right * 2 + $submenu-icon-area-size;
+        .cms-submenu-item {
+            a {
+                position: relative;
+                line-height: $cms-dropdown-item-height;
+                height: $cms-dropdown-item-height;
+                color: $gray-light;
+                padding-left: $cms-dropdown-item-icon-space;
+                &[data-rel] {
```
Comment (FinalAngel) [inline cms/static/cms/sass/components/_subnav.scss]: change the data icon

File: cms/static/cms/sass/components/_subnav.scss [hunk 4]
```diff
@@ -98,6 +134,103 @@
...
+                &[data-icon=copy] {
+                    @include icon(copy);
+                }
+                &[data-icon=scissors] {
+                    @include icon(scissors);
+                }
+                &[data-icon=bin] {
+                    @include icon(bin);
+                }
+                &[href*=alias_plugin] {
```
Comment (FinalAngel) [inline cms/static/cms/sass/components/_subnav.scss]: create task to implement this with the backend

File: cms/static/cms/sass/components/_subnav.scss [hunk 5]
```diff
@@ -98,6 +134,103 @@
...
+        &:before {
+            z-index: z(below);
+            position: absolute;
+            content: '';
+            left: 100%;
+            width: $cms-dropdown-arrow-side;
+            height: $cms-dropdown-arrow-side;
+            margin-left: -($cms-dropdown-arrow-side / 2);
+            transform: rotate(45deg);
+            background-color: white;
```
Comment (FinalAngel) [inline cms/static/cms/sass/components/_subnav.scss]: should be variable and ordering

File: cms/static/cms/sass/components/_subnav.scss [hunk 6]
```diff
@@ -98,6 +134,103 @@
...
+            box-shadow: $cms-dropdown-shadow;
+        }
+
+        &::-webkit-scrollbar {
+            width: 0px;
```
Comment (FinalAngel) [inline cms/static/cms/sass/components/_subnav.scss]: 0 and not 0px

**Comment by FinalAngel** [comment]
nice work :+1:

File: cms/static/cms/js/modules/cms.plugins.js (hunk 1 above)
Comment (vxsx) [inline cms/static/cms/js/modules/cms.plugins.js]: will update in the next PR, as it's going to be quite a refactoring

**Comment by FinalAngel** [comment]
LGTM

**Pull request merged**